src/components/PostDetail.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+<div>
+    <router-Link :to="{ name: 'RenderBlogWithPage', params: {activePage} }" ><button>Go back</button></router-link>
+    
+    <div v-if = "checkIds">
+    <h1> {{currentPost.id}}-  {{ currentPost.title }} </h1>
+    <p> {{currentPost.body}} </p>
+    </div>
+    <h1 v-else > Loading... </h1>
+    </div>
+</template>
+
+<script>
+import axios from "axios"
+
+export default {
+    name: "PostDetail",
+    props: ["postId","activePage"],
+    data(){
+        return {
+            currentPost:{},
+            errorOcurred:false
+        }
+    }, 
+    
+    mounted(){
+        this.makeRequest();
+    },
+
+    methods: {
+        makeRequest(){
+            axios(`https:/jsonplaceholder.typicode.com/posts/${this.postId}`)
+            .then(response => {this.currentPost = response.data})    
+            .catch(() => this.errorOcurred=true);
+        }
+    },
+    
+    computed: {
+        checkIds(){
+          return this.postId==this.currentPost.id;
+        },
+    },
+    watch:{
+        checkError(){
+        alert("boom!  errorr tis an errorr");
+        }
+    }
```

Comment:
@ChoudharyHamza when is this `watch` triggered?

Reply:
oh i meant it to tiggere when a porps changes and that prop was errorOccured it was changed when an API call will return false. but i forgot to add the value on which the change will trigger